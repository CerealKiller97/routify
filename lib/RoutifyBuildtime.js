import { Routify } from '../common/Routify.js'
import { bundlerPlugin } from '../plugins/bundler/index.js'
import { exporterPlugin } from '../plugins/exporter/index.js'
import { filemapperPlugin } from '../plugins/filemapper/index.js'
import { metaFromFilePlugin } from '../plugins/metaFromFile/index.js'

export class RoutifyBuildtime extends Routify {
    plugins = [
        filemapperPlugin,
        metaFromFilePlugin,
        bundlerPlugin,
        exporterPlugin,
    ]
}
