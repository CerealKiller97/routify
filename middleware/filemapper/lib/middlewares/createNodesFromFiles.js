import { Node } from "../../../../lib/node.js";
import { readdir } from 'fs/promises'
import { relative, resolve } from "path";
import { File } from "../File.js";

/**
 * Maps filestructure to a node tree
 * @param {Node} firstNode the root node of the specified path
 * @param {String} path dir to scan for files
 * @return {Promise<Node>}
 */
export async function createNodesFromFiles(firstNode, path) {
    const { instance } = firstNode
    firstNode.file = new File(path)
    const queue = [firstNode]


    let node

    while (node = queue.pop()) {
        node.file.relative = relative(firstNode.file.path, node.file.path) // todo is not including filename
        node.id = node.file.relative.replace(/[^\w]/g, '_')
        
        if (node.file.stat.isDirectory()) {
            // for each child create a node, attach a file and add it to the queue
            const children = await readdir(node.file.path)
            const promises = children.map(filename => {
                const file = new File(resolve(node.file.path, filename))
                const childNode = instance.createNode(file.name)
                childNode.file = file
                node.appendChild(childNode)
                queue.unshift(childNode)
            })
            await Promise.all(promises)
        }

    }
    return firstNode
}