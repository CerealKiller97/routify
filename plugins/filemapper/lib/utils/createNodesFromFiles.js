import { RNode } from '../../../../common/RNode.js' //eslint-disable-line
import fse from 'fs-extra'
import { relative, resolve } from 'path'
import { File } from '../File.js'

/**
 * Maps filestructure to a node tree
 * @param {RNode} firstNode the root node of the specified path
 * @param {String} path dir to scan for files
 * @return {Promise<RNode>}
 */
export async function createNodesFromFiles(firstNode, path) {
    firstNode.file = new File(path)
    const queue = [firstNode]

    let node

    while ((node = queue.pop())) {
        const relativePath = relative(node.root.file.path, node.file.path)
        // set id to `default_path_to_file` if rootName is default
        node.id =
            '_' +
            [node.root.rootName, relativePath]
                .filter(Boolean)
                .join('_')
                .replace(/[^\w]/g, '_')

        if (node.file.stat.isDirectory()) {
            // for each child create a node, attach a file and add it to the queue
            const children = await fse.readdir(node.file.path)
            const promises = children.map((filename) => {
                const file = new File(resolve(node.file.path, filename))
                const component = !file.stat.isDirectory() && file.path
                const childNode = node.createChild(file.name, component)
                childNode.file = file
                queue.unshift(childNode)
            })
            await Promise.all(promises)
        }
    }
    return firstNode
}
