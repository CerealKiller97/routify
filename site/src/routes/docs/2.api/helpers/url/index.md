<script>
    import { resolveNode } from '@roxi/routify'
    import FilesViewer from '#cmp/FilesViewer.svelte'
    import Example from '#cmp/Example.svelte'
</script>


The `$url` helper creates URLs from paths and parameters.

<FilesViewer root={resolveNode('../example')} focus="index.svelte" />

---

# URL Example
<Example offset="../example" />
