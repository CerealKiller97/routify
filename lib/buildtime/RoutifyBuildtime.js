import '#root/typedef.js'
import { Routify } from '../common/Routify.js'
import { normalizePlugins, sortPlugins, deepAssign } from '../common/utils.js'
import { bundlerPlugin } from './plugins/bundler/index.js'
import { exporterPlugin } from './plugins/exporter/index.js'
import { filemapperPlugin } from './plugins/filemapper/index.js'
import { metaFromFilePlugin } from './plugins/metaFromFile/index.js'
import { watcherPlugin } from './plugins/watcher/index.js'
import { hookHandler, resolvePlugins, sanitizeConfig, throttle } from './utils.js'
import { configent } from 'configent'
import { metaSplitPlugin } from './plugins/metaSplit/index.js'
import fse from 'fs-extra'

const getDefaults = () => ({
    routifyDir: '.routify',
    clearRoutifyDir: true,
    filemapper: {
        moduleFiles: ['_module.svelte', '_reset.svelte'],
        resetFiles: ['_reset.svelte'],
    },
    routesDir: {
        default: 'src/routes',
    },
    extensions: ['.svelte', '.html', '.md', '.svx'],
    plugins: [
        filemapperPlugin,
        metaFromFilePlugin,
        bundlerPlugin,
        metaSplitPlugin,
        exporterPlugin,
        watcherPlugin,
    ],
    watch: false,
})

/**
 * @extends {Routify<RNodeConstructor>}
 */
export class RoutifyBuildtime extends Routify {
    /** @param {RoutifyBuildtimeOptions} options */
    constructor(options) {
        const config = deepAssign(
            {},
            getDefaults(),
            configent({ name: 'routify' }),
            options,
        )
        super(sanitizeConfig(config))

        // normalize routifyDir
        if (typeof this.options.routesDir === 'string')
            this.options.routesDir = { default: options.routesDir }

        // clear dir
        // todo jest doesn't like this one
        if (options.clearRoutifyDir) fse.removeSync(options.routifyDir)

        const _build = this.#build.bind(this)
        this.build = async () => await throttle(_build)
    }

    on = {
        buildStart: hookHandler(),
        buildComplete: hookHandler(),
        fileAdded: hookHandler(),
        fileRemoved: hookHandler(),
        fileChanged: hookHandler(),
        fileWatcherReady: hookHandler(),
    }

    async #build() {
        const instance = this
        this.on.buildStart.runHooks()
        this.nodeIndex.splice(0)

        /** @type {RoutifyBuildtimePlugin[]} */
        const plugins = normalizePlugins(await resolvePlugins(this.options.plugins || []))
        this.plugins = sortPlugins(plugins)
        for (const plugin of this.plugins) {
            const shouldRun =
                typeof plugin.condition === 'undefined' ||
                (await plugin.condition({ instance }))
            if (shouldRun) {
                const { options } = plugin
                await plugin.build({ instance, options })
            }
        }

        this.on.buildComplete.runHooks()
    }

    start = this.#build
}
