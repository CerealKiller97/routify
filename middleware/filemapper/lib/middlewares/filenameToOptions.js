import { Node } from "../../../../lib/Node.js"
import { nameFilter } from "../../utils.js";

/**
 * assign reset, fallback and dynamic paths to node.meta
 * @param {Node} node 
 */
export const filenameToOptions = (node) => {
    const { descendants } = node

    // set meta.reset for _reset.svelte
    descendants
        .filter(nameFilter(['_reset.svelte']))
        .forEach(node => node.meta.reset = true)

    // set meta.fallback for _fallback.svelte
    descendants
        .filter(nameFilter(['_fallback.svelte']))
        .forEach(node => node.meta.fallback = true)

    // set meta.dynamic for files with attribute in name
    descendants
        .filter(node => node.file && node.file.name.match(/\[.+\]/))
        .forEach(node => node.meta.dynamic = true)
}