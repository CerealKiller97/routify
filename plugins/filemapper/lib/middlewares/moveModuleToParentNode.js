import { Node } from "../../../../lib/Node.js"
import { nameFilter } from "../../utils.js";


/**
 * for nodes that have a _module.svelte or _reset.svelte file,
 * `file` prop is moved to node.parent and the old node is removed
 * @param {Node} node 
 */
export const moveModuleToParentNode = (node) => {
    const { options } = node.instance
    const { moduleFiles } = options.filemapper

    node.descendants
        .filter(nameFilter(moduleFiles))
        .forEach(node => {
            node.parent.file = node.file
            node.parent.meta = { ...node.parent.meta, ...node.meta }
            node.remove()
        })
}