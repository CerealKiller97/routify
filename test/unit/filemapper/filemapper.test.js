import { suite } from 'uvu'
import * as assert from 'uvu/assert'
import { dirname } from 'path'
import { fileURLToPath } from 'url'
import { createNodesFromFiles } from '../../../plugins/filemapper/lib/utils/createNodesFromFiles.js'
import { moveModuleToParentNode } from '../../../plugins/filemapper/lib/utils/moveModuleToParentNode.js'
import { filenameToOptions } from '../../../plugins/filemapper/lib/utils/filenameToOptions.js'
import { readFileSync, writeFileSync } from 'fs'
import { filemapper } from '../../../plugins/filemapper/lib/index.js'
import { RoutifyBuildtime } from '../../../lib/RoutifyBuildtime.js'

const test = suite('filemapper')
const __dirname = dirname(fileURLToPath(import.meta.url))

const options = {
    filemapper: {
        moduleFiles: ['_module.svelte', '_reset.svelte'],
        resetFiles: ['_reset.svelte'],
        routesDir: { default: `${__dirname}/example` },
    },
}

const instance = new RoutifyBuildtime(options)

let rootNode = instance.createNode()
rootNode.rootName = 'default'

test('files are mapped', async () => {
    await createNodesFromFiles(rootNode, options.filemapper.routesDir.default)
    assert.equal(instance.nodeIndex.length, 13)
    assertSnapshot('1.filemap-only', rootNode, 0)
})

test('modules are merged with parent node', async () => {
    moveModuleToParentNode(rootNode)
    assert.equal(instance.nodeIndex.length, 11)
    assertSnapshot('2.filemap-with-modules', rootNode, 0)
})

test('options get added', async () => {
    filenameToOptions(rootNode)
    assertSnapshot('3.filemap-with-resets', rootNode, 0)
})

test('filemapper', async () => {
    const instance = new RoutifyBuildtime(options)
    await filemapper({ instance })
    assertSnapshot(
        '4.filemap-with-components',
        instance.superNode.children[0],
        0,
    )
})

test.run()

function assertSnapshot(name, content, update) {
    content = JSON.parse(JSON.stringify(content))
    const filepath = `${__dirname}/fixtures/${name}.json`
    if (update) writeFileSync(filepath, JSON.stringify(content, null, 2))
    const expect = JSON.parse(readFileSync(filepath, 'utf-8'))
    assert.equal(content, expect)
}
