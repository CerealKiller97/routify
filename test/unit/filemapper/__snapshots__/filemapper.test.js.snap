// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filemapper: 4.filemap-with-components 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "file": File {
        "base": "_reset.svelte",
        "dir": "test/unit/filemapper/example/admin",
        "ext": ".svelte",
        "name": "_reset",
        "path": "test/unit/filemapper/example/admin/_reset.svelte",
      },
      "id": "_default_admin",
      "meta": Object {
        "reset": true,
      },
      "module": "test/unit/filemapper/example/admin/_reset.svelte::_EVAL",
      "name": "admin",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "file": File {
            "base": "[slug].svelte",
            "dir": "test/unit/filemapper/example/blog",
            "ext": ".svelte",
            "name": "[slug]",
            "path": "test/unit/filemapper/example/blog/[slug].svelte",
          },
          "id": "_default_blog__slug__svelte",
          "meta": Object {
            "dynamic": true,
          },
          "module": "test/unit/filemapper/example/blog/[slug].svelte::_EVAL",
          "name": "[slug]",
        },
        Object {
          "children": Array [],
          "file": File {
            "base": "prefixed[slug].svelte",
            "dir": "test/unit/filemapper/example/blog",
            "ext": ".svelte",
            "name": "prefixed[slug]",
            "path": "test/unit/filemapper/example/blog/prefixed[slug].svelte",
          },
          "id": "_default_blog_prefixed_slug__svelte",
          "meta": Object {
            "dynamic": true,
          },
          "module": "test/unit/filemapper/example/blog/prefixed[slug].svelte::_EVAL",
          "name": "prefixed[slug]",
        },
      ],
      "file": File {
        "base": "blog",
        "dir": "test/unit/filemapper/example",
        "ext": "",
        "name": "blog",
        "path": "test/unit/filemapper/example/blog",
      },
      "id": "_default_blog",
      "meta": Object {},
      "module": "false::_EVAL",
      "name": "blog",
    },
    Object {
      "children": Array [],
      "file": File {
        "base": "index.svelte",
        "dir": "test/unit/filemapper/example",
        "ext": ".svelte",
        "name": "index",
        "path": "test/unit/filemapper/example/index.svelte",
      },
      "id": "_default_index_svelte",
      "meta": Object {},
      "module": "test/unit/filemapper/example/index.svelte::_EVAL",
      "name": "index",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "file": File {
                "base": "foo.svelte",
                "dir": "test/unit/filemapper/example/no-layouts/nested",
                "ext": ".svelte",
                "name": "foo",
                "path": "test/unit/filemapper/example/no-layouts/nested/foo.svelte",
              },
              "id": "_default_no_layouts_nested_foo_svelte",
              "meta": Object {},
              "module": "test/unit/filemapper/example/no-layouts/nested/foo.svelte::_EVAL",
              "name": "foo",
            },
            Object {
              "children": Array [],
              "file": File {
                "base": "index.svelte",
                "dir": "test/unit/filemapper/example/no-layouts/nested",
                "ext": ".svelte",
                "name": "index",
                "path": "test/unit/filemapper/example/no-layouts/nested/index.svelte",
              },
              "id": "_default_no_layouts_nested_index_svelte",
              "meta": Object {},
              "module": "test/unit/filemapper/example/no-layouts/nested/index.svelte::_EVAL",
              "name": "index",
            },
          ],
          "file": File {
            "base": "nested",
            "dir": "test/unit/filemapper/example/no-layouts",
            "ext": "",
            "name": "nested",
            "path": "test/unit/filemapper/example/no-layouts/nested",
          },
          "id": "_default_no_layouts_nested",
          "meta": Object {},
          "module": "false::_EVAL",
          "name": "nested",
        },
      ],
      "file": File {
        "base": "no-layouts",
        "dir": "test/unit/filemapper/example",
        "ext": "",
        "name": "no-layouts",
        "path": "test/unit/filemapper/example/no-layouts",
      },
      "id": "_default_no_layouts",
      "meta": Object {},
      "module": "false::_EVAL",
      "name": "no-layouts",
    },
  ],
  "file": File {
    "base": "_module.svelte",
    "dir": "test/unit/filemapper/example",
    "ext": ".svelte",
    "name": "_module",
    "path": "test/unit/filemapper/example/_module.svelte",
  },
  "id": "_default",
  "meta": Object {},
  "module": "test/unit/filemapper/example/_module.svelte::_EVAL",
  "name": undefined,
  "rootName": "default",
}
`;

exports[`files are mapped: 1.filemap-only 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "file": File {
        "base": "_module.svelte",
        "dir": "test/unit/filemapper/example",
        "ext": ".svelte",
        "name": "_module",
        "path": "test/unit/filemapper/example/_module.svelte",
      },
      "id": "_default__module_svelte",
      "meta": Object {},
      "module": "test/unit/filemapper/example/_module.svelte::_EVAL",
      "name": "_module",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "file": File {
            "base": "_reset.svelte",
            "dir": "test/unit/filemapper/example/admin",
            "ext": ".svelte",
            "name": "_reset",
            "path": "test/unit/filemapper/example/admin/_reset.svelte",
          },
          "id": "_default_admin__reset_svelte",
          "meta": Object {},
          "module": "test/unit/filemapper/example/admin/_reset.svelte::_EVAL",
          "name": "_reset",
        },
      ],
      "file": File {
        "base": "admin",
        "dir": "test/unit/filemapper/example",
        "ext": "",
        "name": "admin",
        "path": "test/unit/filemapper/example/admin",
      },
      "id": "_default_admin",
      "meta": Object {},
      "module": "false::_EVAL",
      "name": "admin",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "file": File {
            "base": "[slug].svelte",
            "dir": "test/unit/filemapper/example/blog",
            "ext": ".svelte",
            "name": "[slug]",
            "path": "test/unit/filemapper/example/blog/[slug].svelte",
          },
          "id": "_default_blog__slug__svelte",
          "meta": Object {},
          "module": "test/unit/filemapper/example/blog/[slug].svelte::_EVAL",
          "name": "[slug]",
        },
        Object {
          "children": Array [],
          "file": File {
            "base": "prefixed[slug].svelte",
            "dir": "test/unit/filemapper/example/blog",
            "ext": ".svelte",
            "name": "prefixed[slug]",
            "path": "test/unit/filemapper/example/blog/prefixed[slug].svelte",
          },
          "id": "_default_blog_prefixed_slug__svelte",
          "meta": Object {},
          "module": "test/unit/filemapper/example/blog/prefixed[slug].svelte::_EVAL",
          "name": "prefixed[slug]",
        },
      ],
      "file": File {
        "base": "blog",
        "dir": "test/unit/filemapper/example",
        "ext": "",
        "name": "blog",
        "path": "test/unit/filemapper/example/blog",
      },
      "id": "_default_blog",
      "meta": Object {},
      "module": "false::_EVAL",
      "name": "blog",
    },
    Object {
      "children": Array [],
      "file": File {
        "base": "index.svelte",
        "dir": "test/unit/filemapper/example",
        "ext": ".svelte",
        "name": "index",
        "path": "test/unit/filemapper/example/index.svelte",
      },
      "id": "_default_index_svelte",
      "meta": Object {},
      "module": "test/unit/filemapper/example/index.svelte::_EVAL",
      "name": "index",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "file": File {
                "base": "foo.svelte",
                "dir": "test/unit/filemapper/example/no-layouts/nested",
                "ext": ".svelte",
                "name": "foo",
                "path": "test/unit/filemapper/example/no-layouts/nested/foo.svelte",
              },
              "id": "_default_no_layouts_nested_foo_svelte",
              "meta": Object {},
              "module": "test/unit/filemapper/example/no-layouts/nested/foo.svelte::_EVAL",
              "name": "foo",
            },
            Object {
              "children": Array [],
              "file": File {
                "base": "index.svelte",
                "dir": "test/unit/filemapper/example/no-layouts/nested",
                "ext": ".svelte",
                "name": "index",
                "path": "test/unit/filemapper/example/no-layouts/nested/index.svelte",
              },
              "id": "_default_no_layouts_nested_index_svelte",
              "meta": Object {},
              "module": "test/unit/filemapper/example/no-layouts/nested/index.svelte::_EVAL",
              "name": "index",
            },
          ],
          "file": File {
            "base": "nested",
            "dir": "test/unit/filemapper/example/no-layouts",
            "ext": "",
            "name": "nested",
            "path": "test/unit/filemapper/example/no-layouts/nested",
          },
          "id": "_default_no_layouts_nested",
          "meta": Object {},
          "module": "false::_EVAL",
          "name": "nested",
        },
      ],
      "file": File {
        "base": "no-layouts",
        "dir": "test/unit/filemapper/example",
        "ext": "",
        "name": "no-layouts",
        "path": "test/unit/filemapper/example/no-layouts",
      },
      "id": "_default_no_layouts",
      "meta": Object {},
      "module": "false::_EVAL",
      "name": "no-layouts",
    },
  ],
  "file": File {
    "base": "example",
    "dir": "test/unit/filemapper",
    "ext": "",
    "name": "example",
    "path": "test/unit/filemapper/example",
  },
  "id": "_default",
  "meta": Object {},
  "module": "undefined::_EVAL",
  "name": undefined,
  "rootName": "default",
}
`;

exports[`modules are merged with parent node: 2.filemap-with-modules 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "file": File {
        "base": "_reset.svelte",
        "dir": "test/unit/filemapper/example/admin",
        "ext": ".svelte",
        "name": "_reset",
        "path": "test/unit/filemapper/example/admin/_reset.svelte",
      },
      "id": "_default_admin",
      "meta": Object {},
      "module": "test/unit/filemapper/example/admin/_reset.svelte::_EVAL",
      "name": "admin",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "file": File {
            "base": "[slug].svelte",
            "dir": "test/unit/filemapper/example/blog",
            "ext": ".svelte",
            "name": "[slug]",
            "path": "test/unit/filemapper/example/blog/[slug].svelte",
          },
          "id": "_default_blog__slug__svelte",
          "meta": Object {},
          "module": "test/unit/filemapper/example/blog/[slug].svelte::_EVAL",
          "name": "[slug]",
        },
        Object {
          "children": Array [],
          "file": File {
            "base": "prefixed[slug].svelte",
            "dir": "test/unit/filemapper/example/blog",
            "ext": ".svelte",
            "name": "prefixed[slug]",
            "path": "test/unit/filemapper/example/blog/prefixed[slug].svelte",
          },
          "id": "_default_blog_prefixed_slug__svelte",
          "meta": Object {},
          "module": "test/unit/filemapper/example/blog/prefixed[slug].svelte::_EVAL",
          "name": "prefixed[slug]",
        },
      ],
      "file": File {
        "base": "blog",
        "dir": "test/unit/filemapper/example",
        "ext": "",
        "name": "blog",
        "path": "test/unit/filemapper/example/blog",
      },
      "id": "_default_blog",
      "meta": Object {},
      "module": "false::_EVAL",
      "name": "blog",
    },
    Object {
      "children": Array [],
      "file": File {
        "base": "index.svelte",
        "dir": "test/unit/filemapper/example",
        "ext": ".svelte",
        "name": "index",
        "path": "test/unit/filemapper/example/index.svelte",
      },
      "id": "_default_index_svelte",
      "meta": Object {},
      "module": "test/unit/filemapper/example/index.svelte::_EVAL",
      "name": "index",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "file": File {
                "base": "foo.svelte",
                "dir": "test/unit/filemapper/example/no-layouts/nested",
                "ext": ".svelte",
                "name": "foo",
                "path": "test/unit/filemapper/example/no-layouts/nested/foo.svelte",
              },
              "id": "_default_no_layouts_nested_foo_svelte",
              "meta": Object {},
              "module": "test/unit/filemapper/example/no-layouts/nested/foo.svelte::_EVAL",
              "name": "foo",
            },
            Object {
              "children": Array [],
              "file": File {
                "base": "index.svelte",
                "dir": "test/unit/filemapper/example/no-layouts/nested",
                "ext": ".svelte",
                "name": "index",
                "path": "test/unit/filemapper/example/no-layouts/nested/index.svelte",
              },
              "id": "_default_no_layouts_nested_index_svelte",
              "meta": Object {},
              "module": "test/unit/filemapper/example/no-layouts/nested/index.svelte::_EVAL",
              "name": "index",
            },
          ],
          "file": File {
            "base": "nested",
            "dir": "test/unit/filemapper/example/no-layouts",
            "ext": "",
            "name": "nested",
            "path": "test/unit/filemapper/example/no-layouts/nested",
          },
          "id": "_default_no_layouts_nested",
          "meta": Object {},
          "module": "false::_EVAL",
          "name": "nested",
        },
      ],
      "file": File {
        "base": "no-layouts",
        "dir": "test/unit/filemapper/example",
        "ext": "",
        "name": "no-layouts",
        "path": "test/unit/filemapper/example/no-layouts",
      },
      "id": "_default_no_layouts",
      "meta": Object {},
      "module": "false::_EVAL",
      "name": "no-layouts",
    },
  ],
  "file": File {
    "base": "_module.svelte",
    "dir": "test/unit/filemapper/example",
    "ext": ".svelte",
    "name": "_module",
    "path": "test/unit/filemapper/example/_module.svelte",
  },
  "id": "_default",
  "meta": Object {},
  "module": "test/unit/filemapper/example/_module.svelte::_EVAL",
  "name": undefined,
  "rootName": "default",
}
`;

exports[`options get added: 3.filemap-with-resets 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "file": File {
        "base": "_reset.svelte",
        "dir": "test/unit/filemapper/example/admin",
        "ext": ".svelte",
        "name": "_reset",
        "path": "test/unit/filemapper/example/admin/_reset.svelte",
      },
      "id": "_default_admin",
      "meta": Object {
        "reset": true,
      },
      "module": "test/unit/filemapper/example/admin/_reset.svelte::_EVAL",
      "name": "admin",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "file": File {
            "base": "[slug].svelte",
            "dir": "test/unit/filemapper/example/blog",
            "ext": ".svelte",
            "name": "[slug]",
            "path": "test/unit/filemapper/example/blog/[slug].svelte",
          },
          "id": "_default_blog__slug__svelte",
          "meta": Object {
            "dynamic": true,
          },
          "module": "test/unit/filemapper/example/blog/[slug].svelte::_EVAL",
          "name": "[slug]",
        },
        Object {
          "children": Array [],
          "file": File {
            "base": "prefixed[slug].svelte",
            "dir": "test/unit/filemapper/example/blog",
            "ext": ".svelte",
            "name": "prefixed[slug]",
            "path": "test/unit/filemapper/example/blog/prefixed[slug].svelte",
          },
          "id": "_default_blog_prefixed_slug__svelte",
          "meta": Object {
            "dynamic": true,
          },
          "module": "test/unit/filemapper/example/blog/prefixed[slug].svelte::_EVAL",
          "name": "prefixed[slug]",
        },
      ],
      "file": File {
        "base": "blog",
        "dir": "test/unit/filemapper/example",
        "ext": "",
        "name": "blog",
        "path": "test/unit/filemapper/example/blog",
      },
      "id": "_default_blog",
      "meta": Object {},
      "module": "false::_EVAL",
      "name": "blog",
    },
    Object {
      "children": Array [],
      "file": File {
        "base": "index.svelte",
        "dir": "test/unit/filemapper/example",
        "ext": ".svelte",
        "name": "index",
        "path": "test/unit/filemapper/example/index.svelte",
      },
      "id": "_default_index_svelte",
      "meta": Object {},
      "module": "test/unit/filemapper/example/index.svelte::_EVAL",
      "name": "index",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "file": File {
                "base": "foo.svelte",
                "dir": "test/unit/filemapper/example/no-layouts/nested",
                "ext": ".svelte",
                "name": "foo",
                "path": "test/unit/filemapper/example/no-layouts/nested/foo.svelte",
              },
              "id": "_default_no_layouts_nested_foo_svelte",
              "meta": Object {},
              "module": "test/unit/filemapper/example/no-layouts/nested/foo.svelte::_EVAL",
              "name": "foo",
            },
            Object {
              "children": Array [],
              "file": File {
                "base": "index.svelte",
                "dir": "test/unit/filemapper/example/no-layouts/nested",
                "ext": ".svelte",
                "name": "index",
                "path": "test/unit/filemapper/example/no-layouts/nested/index.svelte",
              },
              "id": "_default_no_layouts_nested_index_svelte",
              "meta": Object {},
              "module": "test/unit/filemapper/example/no-layouts/nested/index.svelte::_EVAL",
              "name": "index",
            },
          ],
          "file": File {
            "base": "nested",
            "dir": "test/unit/filemapper/example/no-layouts",
            "ext": "",
            "name": "nested",
            "path": "test/unit/filemapper/example/no-layouts/nested",
          },
          "id": "_default_no_layouts_nested",
          "meta": Object {},
          "module": "false::_EVAL",
          "name": "nested",
        },
      ],
      "file": File {
        "base": "no-layouts",
        "dir": "test/unit/filemapper/example",
        "ext": "",
        "name": "no-layouts",
        "path": "test/unit/filemapper/example/no-layouts",
      },
      "id": "_default_no_layouts",
      "meta": Object {},
      "module": "false::_EVAL",
      "name": "no-layouts",
    },
  ],
  "file": File {
    "base": "_module.svelte",
    "dir": "test/unit/filemapper/example",
    "ext": ".svelte",
    "name": "_module",
    "path": "test/unit/filemapper/example/_module.svelte",
  },
  "id": "_default",
  "meta": Object {},
  "module": "test/unit/filemapper/example/_module.svelte::_EVAL",
  "name": undefined,
  "rootName": "default",
}
`;
